pkg/test/echo/response/check/checker.go

```diff
@@ -0,0 +1,46 @@
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package check
+
+import (
+	"fmt"
+
+	"istio.io/istio/pkg/test/echo/response"
+	"istio.io/istio/pkg/test/echo/response/checkcall"
+)
+
+// Checker inspects echo responses for errors.
+type Checker interface {
```

Comment:
I am not sure I understand the point of this interface. It seems all usages are checkcall. I understand the distinction (some checks can apply to a single request (and therefor also all requests) vs others need all requests) but it seems like an internal distinction. 

This is more confusing when we have the same method in both. I wouldn't know how to use check.And vs checkall.And, and I suspect many others would not.

Would it make sense to consolidate these to a single `check.Checker` interface which acts on response.Instances? For the checkers that act on only one response at a time that can be abstracted out with a common (private) function that upconverts single checkers to multi checkers.

Not sure that makes sense - LMK what you think

Reply:
I waffled back and forth on this for a while as well. The main motivation for the `checker` package was failing fast on messages, but I'm not actually sure how much of an optimization that is in practice. Let me try merging the two as you suggested.